dask/dataframe/core.py

```diff
@@ -75,12 +75,23 @@ def _get_return_type(meta):
     return Scalar
 
 
-def new_dd_object(dsk, _name, meta, divisions):
+def new_dd_object(dsk, name, meta, divisions):
     """Generic constructor for dask.dataframe objects.
 
     Decides the appropriate output class based on the type of `meta` provided.
     """
-    return _get_return_type(meta)(dsk, _name, meta, divisions)
+    if isinstance(meta, np.ndarray):
```

Comment:
It would be useful to generalize this beyond numpy arrays.  Do we have a good test for duck arrays?  cc @shoyer 

test to see if they have `__array_ufunc__`?  Test to see if they have shape and dtype?  (and aren't also pandas dataframes)

Reply:
Also cc @jcrist who determined the policy for this on the dask.array side